lighthouse-core/audits/seo/structured-data-automatic.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit.js');
+const validateJsonLD = require('../../lib/sd-validation/sd-validation.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on whether JSON-LD structured data snippets are valid. This descriptive title is shown when no invalid JSON-LD snippets were found. */
+  title: 'JSON-LD structured data syntax is valid',
+  /** Title of a Lighthouse audit that provides detail on whether JSON-LD structured data snippets are valid. This descriptive title is shown when JSON-LD snippets with invalid content were found. */
+  failureTitle: 'JSON-LD structured data syntax is invalid',
+  /** Description of a Lighthouse audit that tells the user whether JSON-LD snippets on the page are invalid. This is displayed after a user expands the section to see more. No character length limits. */
+  /* eslint-disable-next-line max-len */
+  description: 'Structured data contains rich metadata about a web page. The data is used in search results and social sharing. Invalid metadata will affect how the page appears in these contexts. This audit currently validates a subset of JSON-LD rules. See also the manual audit below to learn how to validate other types of structured data.',
```

Comment:
I wonder if we should just include the link to the tool here rather than force them to go to the other audit?

Reply:
This is the message for the manual structured data audit:

> Run the [Structured Data Testing Tool](https://search.google.com/structured-data/testing-tool/) and the [Structured Data Linter](http://linter.structured-data.org/) to validate structured data. [Learn more](https://developers.google.com/search/docs/guides/mark-up-content?utm_source=lighthouse&utm_medium=cli).

The Structured Data Linter doesn't look very welcoming, so maybe we can remove it and just link the SDTT from both audits.

Do we even need the manual audit? If we removed it we'd need to change the title of the automatic audit to not only mention JSON-LD. But maybe that's misleading, if we say your structured data is fine but we only checked JSON-LD?

@avgp any thoughts on this?

(Related: should we call the audit `structured-data-automatic` internally or just `json-ld`?)

Reply:
I'll defer to @AVGP for removal of the manual one. `json-ld` seems like a reasonable audit id to me. Do we foresee this growing in scope to include other types of data?

Reply:
Renamed the audit to `json-ld`.

Will bring up description/manual audit with @avgp again in a week or two, but shouldn't be merge blocking.

Reply:
[Updates](https://github.com/GoogleChrome/lighthouse/pull/8328/commits/991bf3c0e30e201224d3956fba4e65a1cb0b159a) after talking to @avgp:

- removed the link to the Structured Data Linter
- link to SDTT from json-ld audit description

Reply:
Yep, I think there's value in having the manual audit as people get confused with "valid JSON-LD" vs. "Valid SD for rich results" and such, plus the SDTT catches most of that, so linking directly to it is a good idea.